### DISCUSSION TITLE
WebView2 as a replacement for the old IE browser control?

### DISCUSSION THREAD

**Discussion by m8e** [twinbasic/twinbasic#762] [Ideas]
Some kind of browser automation would be useful in TwinBasic

I created many VBx web scrapers over the years starting as far back as VB5
(and then later using Access VBA instantiating an Internet Explorer using WithEvents to react to the async callbacks)

I also used the Web Browser control in Access frontends to help transition parts of a (massive) SQL Server app to mobile-web 

i.e. I created a NGINX+PHP server as the new View+Controller layers (talking to our existing SQL Server) that get embedded in the Access frontend via Web Browser Control, and the PHP gets re-used externally in a mobile app wrapper for the field workers who are outside the headquarters building)

Anyway, I don't expect (or need) TwinBasic to convert my IE browser automation code, but I was wondering about the future ideas regarding browser automation with TwinBasic?

WebView2 looks the obvious answer to replace the old IE-based programmable browser object
https://docs.microsoft.com/en-us/microsoft-edge/webview2/get-started/get-started

By 2023 the WebView2 should be inside the newest version of MS Access:
https://www.microsoft.com/en-us/microsoft-365/roadmap?filters=Access

**Comment by mburns08109** [comment]
I would love to take a look at some of your browser scraping VBA code sometime.

**Reply by m8e** [reply]
@mburns08109 will see what I can dredge up from ye olde archives next time I am inside my BackBlaze control panel :) 

The best VBA scraper I wrote was a web macro recorder as well, Access launched a separate WithEvents "watched" instance of IE to respond to a handful of particular events emitted by the browser, and record only URLs with POST and GET parameters editable for custom scripting/scraping re-runs.  

Writing a self-contained recorder-scraper-bot in Access97 was a bit of fun back in the day.   

I think it worked up to about Access 2003 and stopped working after IE5 got more security bolted on top (or was it IE6?)

I briefly attempted ressurrecting it about 5 years ago using some Firefox scriptable wrapper but that stopped working so I decided PHP and Python were more stable way going forward.   My other speed tool in my kit is Rtila.net robot-maker (I'm one of their early adopters, using it for custom server monitoring and service reboots etc)

Automatable Chrome/WebView2 looks great as a broadly supported API moving torwards 2030

**Comment by mburns08109** [comment]
Though, this having been said, won't we HAVE TO support the OLD IE control natively - for that 100% backwards compatibility thing (even though it's "no longer supported" and all?

**Reply by bclothier** [reply]
Not directly. It's just an ActiveX control. if we can host ActiveX control, we can host the old IE control and indeed any other old ActiveX controls from the 1890s if one so insist. ;-)
